### DISCUSSION TITLE
PCPCombine: FCST_PCP_COMBINE_INPUT_OPTIONS not being used?

### DISCUSSION THREAD

**Discussion by malloryprow** [dtcenter/METplus#2158] [Configuration] [answered]
In the NAM, there are two "WEASD/Z0" variables

> wgrib2 /lfs/h1/ops/prod/com/nam/v4.2/nam.20230430/nam.t06z.awip1206.tm00.grib2 | grep "WEASD"
> 64:10693511:d=2023043006:WEASD:surface:6 hour fcst:
> 89:14150378:d=2023043006:WEASD:surface:3-6 hour acc fcst:

They different in 

> 64:10693511:code table 4.0=0 Analysis or forecast at a horizontal level or in a horizontal layer at a point in time
> 89:14150378:code table 4.0=8 Average, accumulation, extreme values or other statistically processed values at a horizontal level or in a horizontal layer in a continuous or non-continuous time interval

I'd like to use record 64, so in my METplus conf file I set _FCST_PCP_COMBINE_INPUT_OPTIONS = GRIB2_pdt=0;_. However, it doesn't look the PCPCombine is using this information because the command that ran was 

/apps/ops/para/libs/intel/19.1.3.304/met/11.0.1/bin/pcp_combine -v 2 -subtract /lfs/h1/ops/prod/com/nam/v4.2/nam.20230430/nam.t06z.awip1206.tm00.grib2 'name="WEASD"; level="Z0";' /lfs/h1/ops/prod/com/nam/v4.2/nam.20230430/nam.t06z.awip1200.tm00.grib2 'name="WEASD"; level="Z0";' -name "WEASD_A06" pcp_combine_nam_accum06hr_WEASD_init2023043006_fhr006.nc

but I would have expected

/apps/ops/para/libs/intel/19.1.3.304/met/11.0.1/bin/pcp_combine -v 2 -subtract /lfs/h1/ops/prod/com/nam/v4.2/nam.20230430/nam.t06z.awip1206.tm00.grib2 'name="WEASD"; level="Z0"; GRIB2_pdt=0;' /lfs/h1/ops/prod/com/nam/v4.2/nam.20230430/nam.t06z.awip1200.tm00.grib2 'name="WEASD"; level="Z0"; GRIB2_pdt=0;' -name WEASD_A06 pcp_combine_nam_accum06hr_WEASD_init2023043006_fhr006.nc

**Comment by georgemccabe** [comment]
Hi @malloryprow, I confirmed that I get the same behavior. It looks like there is a bug in the PCPCombine wrapper when running in -subtract mode. I created issue #2161 to track this fix. Do you need a bugfix to be applied to METplus 5.0?

**Reply by malloryprow** [reply]
Nope! I was able to use USER_DEFINED to get what I needed done. Thanks George!

**Reply by georgemccabe** [reply]
Sounds good. I will made this fix for the upcoming METplus 5.1.0 release. Thanks for catching this bug!

**Reply by georgemccabe** [reply, marked as answer]
I created PR #2162 for this fix on the develop branch. Could you review it when the tests finish running?

**Reply by jprestop** [reply]
The PR mentioned above was approved and merged, so I am marking this Discussion as Answered.
